exercises/clock/example_gen.go

```diff
@@ -15,7 +16,8 @@ func main() {
 		log.Fatal(err)
 	}
 	var j js
-	if err := gen.Gen("clock.json", &j, t); err != nil {
+	f := path.Join("exercises", "clock", "canonical-data.json")
```

Comment:
The `path.Join("exercises", whateverslug, "canonical-data.json")` is common to every single `example_gen.go`. How would you feel about moving it into the `Gen` function?

I would like to be able to say `gen.Gen(whateverSlug, &j, t)`, and have it infer the path from the slug. That means further reorganisational changes like https://github.com/exercism/x-common/pull/351 can be dealt with by just changing `Gen`, rather than every single generator.